stdlib/public/runtime/Exclusivity.cpp

```diff
@@ -202,6 +208,7 @@ class AccessSet {
   }
 
   void remove(Access *access) {
+    assert(Head != nullptr && "Can not remove access from empty access list");
```

Comment:
I put in this assert so we always get a nice msg when we are in this invalid state.

Reply:
FYI I have the same assert planned for https://github.com/apple/swift/pull/38215 which is close to being merged (just need to fix a test on Linux).

Reply:
Sounds good! I'll be sure to rebase. I think this is going to take longer.